### ISSUE TITLE
App state set value not working while open as Modal

### ISSUE THREAD

**Issue by ggs331** [Budibase/budibase#3781]
**Describe the bug**
App state set value not working while open as Modal, but works on Full Screen mode.

Please find the screen video. If the page open as modal, the app state set value on the modal page didn't work.
https://user-images.githubusercontent.com/29649651/145947452-519625f3-02cb-4b23-9344-8168ddfadd48.mov

**Comment by ggs331** [comment]
BTW, everything work fine on v0.9.173, this issue just happened  after upgraded to 1.0.x.

**Comment by ggs331** [comment]
When I trying to reproduce this issue, I found something strange. Then `Update State` will not work in modal if it's ran after `Execute Query`, but works in modal if I put the `Update State` before `Execute Query`

If `Update State` as the below screenshot, the `Update State` not work while open as modal.
![image](https://user-images.githubusercontent.com/29649651/145950204-a34181e0-88d4-4cb3-a4fe-cf69dc921e0d.png)

If I put the `Update State` on top of the actions, then it works on both modal and full screen mode.
![image](https://user-images.githubusercontent.com/29649651/145950627-8e4e3778-a64b-4bab-a6f2-5067d87d3f29.png)

**Comment by aptkingston** [comment]
Thank you for the detailed description and the video @ggs331! It makes things much easier to reproduce and fix. I know the reason for this already and think it should be an easy fix.

**Comment by ggs331** [comment]
Thank you @aptkingston . Appreciate if you can let me know when this fix released as I was stuck on 0.9.173 due to this issue.

**Comment by ggs331** [comment]
@aptkingston Thanks for the update, I have upgraded to 1.0.24 and seems the issue mentions before has been fixed. 

But after upgraded just for few hours, I found another issue is: when button click and updated app state, it may caused other button `Navigate To > Open screen in modal` just didn't work(no any response at all), and this open modal not working issue not just happened on a single page, but all of other pages. I still not find how to reproduce this issue. But it happens on my apps. Now I have to disable `Open screen in modal` option temporary. I am wondering this issue maybe something related to this bug fix?

I will update you if I find more details.

**Comment by aptkingston** [comment]
Hi @ggs331. Could you take a screenshot of your button actions? I have tried to reproduce this but everything works.

I have tried:
- Update state only
- Update state, then navigate to a screen in a modal
- Navigate to a screen in a modal, then update state
- Navigate to a screen in a modal only

All of my examples work though. It would really help if you could either provide a screenshots or exact steps that reproduce this. Thanks!

**Comment by ggs331** [comment]
Hi @aptkingston , I have recored a screen video. This open modal issue triggered by app state value update, it caused all my apps pages open modal not working.

When I fist open the page, the button click open modal just working fine.
![image](https://user-images.githubusercontent.com/29649651/146618672-a1c483ff-6e3d-4c39-98cf-87c4dd2d567d.png)

Then I tried to click some buttons, these buttons actually performed app state update value just like the screenshot, then triggered my data provider filter change and refresh. But this action actually caused open modal not working, and this also impacted to other pages's open modal. 
![image](https://user-images.githubusercontent.com/29649651/146618795-0025c3b0-36c3-41d3-83ee-98cf3932ae8b.png)


https://user-images.githubusercontent.com/29649651/146618518-f12f9ef9-437c-4e0b-b116-0e3dcfb75fa0.mov
